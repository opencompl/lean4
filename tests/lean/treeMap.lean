inductive TreeNode where
 | mkLeaf (name : String) : TreeNode
 | mkNode (name : String) (children : List TreeNode) : TreeNode

open TreeNode in def treeToList (t : TreeNode) : List String :=
 match t with
 | mkLeaf name => [name]
 | mkNode name children => name :: List.flatten (children.map treeToList)
termination_by t
