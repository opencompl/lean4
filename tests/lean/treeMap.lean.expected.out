treeMap.lean:8:62-8:72: error: failed to prove termination, possible solutions:
  - Use `have`-expressions to prove the remaining goals
  - Use `termination_by` to specify a different well-founded relation
  - Use `decreasing_by` to specify your own tactic for discharging this kind of goal
name : String
children : List TreeNode
t : TreeNode
‚ä¢ sizeOf t < 1 + sizeOf name + sizeOf children
